compiler/aarch64/codegen/OMRCodeGenerator.cpp

```diff
@@ -256,6 +257,13 @@ OMR::ARM64::CodeGenerator::doBinaryEncoding()
       self()->setBinaryBufferCursor(cursorInstruction->generateBinaryEncoding());
       cursorInstruction = cursorInstruction->getNext();
       }
+
+   TR::Instruction *prologueCursor = self()->getFirstInstruction();
+   for (TR::Instruction *gcMapCursor = prologueCursor; NULL!= gcMapCursor; gcMapCursor = gcMapCursor->getNext())
+      {
+      if (gcMapCursor->needsGCMap())
+         gcMapCursor->setGCMap(self()->getStackAtlas()->getParameterMap()->clone(self()->trMemory()));
+      }
```

Comment:
What are you trying to accomplish with this code?  Unless I'm misunderstanding, this walk should be limited to just a generated method prologue (which cares about parameters).  However, it appears to be walking over all instructions in a method.  If you're going to do this you need to only walk over the start/end instructions in the prologue.  This code should probably live earlier as well, say after L220.

A loop like this is really only needed if the prologue introduces instructions that are GC points for which a map hasn't previously been created.  I'm not sure yet if this will be needed for the AArch64 private linkage, but I think it's safe to leave this in for now.

Reply:
Ok. I've moved it up to just after L220. Let me know if this looks good now.